packages/agreement-process/test/submitAgreement.test.ts

```diff
@@ -0,0 +1,1742 @@
+      fail("Creation fails: agreement not found in event-store");
+    }
+
+    expect(actualAgreementData.type).toEqual("AgreementActivated");
+    expect(actualAgreementData).toMatchObject({
+      type: "AgreementActivated",
+      event_version: 2,
+      version: "1",
+      stream_id: submittedAgreement.id,
+    });
+
+    const actualAgreement: AgreementV2 | undefined = decodeProtobufPayload({
+      messageType: AgreementActivatedV2,
+      payload: actualAgreementData.data,
+    }).agreement;
+
+    if (!actualAgreement) {
+      fail("impossible to decode AgreementAddedV1 data");
+    }
+
+    expect(selfcareV2ClientMock.getUserInfoUsingGET).toHaveBeenCalled();
+    expect(actualAgreement).toMatchObject({
+      id: submittedAgreement.id,
+      eserviceId: eservice.id,
+      descriptorId: descriptor.id,
+      producerId: producer.id,
+      consumerId: consumer.id,
+      state: toAgreementStateV2(agreementState.active),
+      contract: {
+        id: expect.any(String),
+        name: expect.any(String),
+        prettyName: expect.any(String),
+        contentType: expect.any(String),
+        path: expect.any(String),
+        createdAt: expect.any(BigInt),
+      },
+      consumerNotes: consumerNotesText,
+      verifiedAttributes: [
+        {
+          id: validVerifiedTenantAttribute.id,
+        },
+      ],
+      suspendedByConsumer: false,
+      suspendedByProducer: false,
+      suspendedByPlatform: false,
+    });
+    expect(actualAgreement.stamps?.activation).toMatchObject({
+      who: authData.userId,
+      when: expect.any(BigInt),
+    });
+  });
+
+  it("should submit agreement contract with new state ACTIVE when producer and consumer are differents, generates AgreementActivated event and AgreementArchivedByUpgrade for related agreements", async () => {
+    const consumerId = generateId<TenantId>();
+    const producer = getMockTenant(consumerId);
+    const consumerNotesText = "This is a test";
+
+    const validVerifiedTenantAttribute: TenantAttribute = {
+      ...getMockVerifiedTenantAttribute(),
+      verifiedBy: [
+        {
+          id: consumerId,
+          verificationDate: new Date(new Date().getFullYear() - 1),
+          expirationDate: new Date(new Date().getFullYear() + 1),
+          extensionDate: undefined,
+        },
+      ],
+    };
+
+    const consumer = {
+      ...getMockTenant(consumerId, [validVerifiedTenantAttribute]),
+      selfcareId: generateId<SelfcareId>(),
+      mails: [
+        {
+          id: generateId(),
+          kind: tenantMailKind.ContactEmail,
+          address: "avalidemailaddressfortenant@testingagreement.com",
+          createdAt: new Date(),
+        },
+      ],
+    };
+
+    const descriptor = {
+      ...getMockDescriptor(),
+      state: descriptorState.suspended,
+      attributes: {
+        certified: [],
+        declared: [],
+        verified: [[getMockEServiceAttribute(validVerifiedTenantAttribute.id)]],
+      },
```

Comment:
When producer and consumer are different, an agreement is activated only if all certified, verified, and declared attributes are satisfied, so you should add all the three types of attributes here (not sure why this is not failing)

Reply:
The check is equally valid if the descriptor does not contain any attributes. The condition is that the tenant must have the attributes specified in the descriptor, so a descriptor without attributes is always valid.
However, I added the attributes through this commit [64876646b6c866f27815d1fea98f955e1a55fe48](https://github.com/pagopa/interop-be-monorepo/pull/603/commits/64876646b6c866f27815d1fea98f955e1a55fe48).

In this way, the test is more accurate.

Reply:
It seems you added them to a different test, the one where the producer and consumer are the same, but in that case the attributes are not checked - they should be added to this test case to cover the attributes check